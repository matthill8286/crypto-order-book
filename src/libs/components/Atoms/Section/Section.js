function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { media } from "../../../styles/media";
import { css, styled } from "../../../styles/styled";
import { handleShape } from "../Card/Card.styled";

var handlePadding = function handlePadding(paddingTop, paddingBottom) {
  var top = _extends({}, paddingTop);

  var bottom = _extends({}, paddingBottom);

  return function (props) {
    return css(["", ";", ";", "{", ";", ";}", "{", ";", ";}"], top.mobile && "padding-top: ".concat(props.theme.spacing.base[top.mobile]), bottom.mobile && "padding-bottom: ".concat(props.theme.spacing.base[bottom.mobile]), media.md, top.tablet && "padding-top: ".concat(props.theme.spacing.base[top.tablet]), bottom.tablet && "padding-bottom: ".concat(props.theme.spacing.base[bottom.tablet]), media.lg, top.desktop && "padding-top: ".concat(props.theme.spacing.base[top.desktop]), bottom.desktop && "padding-bottom: ".concat(props.theme.spacing.base[bottom.desktop]));
  };
};

export var Section = styled.section.withConfig({
  displayName: "Section",
  componentId: "d5a2e3-0"
})(["width:100%;height:", ";left:0;right:0;background:", " no-repeat center center;background-image:", ";background-size:cover;", " ", " ", " ", "{background-image:", ";height:", ";}", "{background-image:", ";height:", ";}"], function (_ref) {
  var height = _ref.height;
  return height && height.mobile && "".concat(height.mobile, "px");
}, function (_ref2) {
  var color = _ref2.color,
      theme = _ref2.theme;
  return color && theme.color[color];
}, function (_ref3) {
  var image = _ref3.image;
  return image && image.mobile && "url(".concat(image.mobile, ") ");
}, function (_ref4) {
  var paddingTop = _ref4.paddingTop,
      paddingBottom = _ref4.paddingBottom;
  return handlePadding(paddingTop, paddingBottom);
}, function (_ref5) {
  var center = _ref5.center;
  return center && "text-align: center;";
}, function (_ref6) {
  var shape = _ref6.shape;
  return shape && handleShape(shape);
}, media.md, function (_ref7) {
  var image = _ref7.image;
  return image && image.tablet && "url(".concat(image.tablet, ") ");
}, function (_ref8) {
  var height = _ref8.height;
  return height && height.tablet && "".concat(height.tablet, "px");
}, media.lg, function (_ref9) {
  var image = _ref9.image;
  return image && image.desktop && "url(".concat(image.desktop, ") ");
}, function (_ref10) {
  var height = _ref10.height;
  return height && height.desktop && "".concat(height.desktop, "px");
});
//# sourceMappingURL=Section.js.map