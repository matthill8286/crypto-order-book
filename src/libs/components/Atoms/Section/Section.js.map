{"version":3,"sources":["../../../../src/components/Atoms/Section/Section.tsx"],"names":["media","css","styled","handleShape","handlePadding","paddingTop","paddingBottom","top","bottom","props","mobile","theme","spacing","base","md","tablet","lg","desktop","Section","section","height","color","image","center","shape"],"mappings":";;AAAA,SAASA,KAAT;AACA,SAASC,GAAT,EAAcC,MAAd;AAGA,SAASC,WAAT;;AAYA,IAAMC,aAAa,GAAG,SAAhBA,aAAgB,CAACC,UAAD,EAAaC,aAAb,EAA+B;AAAA,MACxCC,GADwC,gBAChCF,UADgC;;AAAA,MAExCG,MAFwC,gBAE7BF,aAF6B;;AAInD,SAAO,UAAAG,KAAK;AAAA,WAAIR,GAAJ,iDACRM,GAAG,CAACG,MAAJ,2BAA8BD,KAAK,CAACE,KAAN,CAAYC,OAAZ,CAAoBC,IAApB,CAAyBN,GAAG,CAACG,MAA7B,CAA9B,CADQ,EAERF,MAAM,CAACE,MAAP,8BAAoCD,KAAK,CAACE,KAAN,CAAYC,OAAZ,CAAoBC,IAApB,CAAyBL,MAAM,CAACE,MAAhC,CAApC,CAFQ,EAIRV,KAAK,CAACc,EAJE,EAKNP,GAAG,CAACQ,MAAJ,2BAA8BN,KAAK,CAACE,KAAN,CAAYC,OAAZ,CAAoBC,IAApB,CAAyBN,GAAG,CAACQ,MAA7B,CAA9B,CALM,EAMNP,MAAM,CAACO,MAAP,8BAAoCN,KAAK,CAACE,KAAN,CAAYC,OAAZ,CAAoBC,IAApB,CAAyBL,MAAM,CAACO,MAAhC,CAApC,CANM,EASRf,KAAK,CAACgB,EATE,EAUNT,GAAG,CAACU,OAAJ,2BAA+BR,KAAK,CAACE,KAAN,CAAYC,OAAZ,CAAoBC,IAApB,CAAyBN,GAAG,CAACU,OAA7B,CAA/B,CAVM,EAWNT,MAAM,CAACS,OAAP,8BAAqCR,KAAK,CAACE,KAAN,CAAYC,OAAZ,CAAoBC,IAApB,CAAyBL,MAAM,CAACS,OAAhC,CAArC,CAXM;AAAA,GAAZ;AAcD,CAlBD;;AAoBA,OAAO,IAAMC,OAAO,GAAGhB,MAAM,CAACiB,OAAV;AAAA;AAAA;AAAA,kOAER;AAAA,MAAGC,MAAH,QAAGA,MAAH;AAAA,SAAgBA,MAAM,IAAIA,MAAM,CAACV,MAAjB,cAA8BU,MAAM,CAACV,MAArC,OAAhB;AAAA,CAFQ,EAKJ;AAAA,MAAGW,KAAH,SAAGA,KAAH;AAAA,MAAUV,KAAV,SAAUA,KAAV;AAAA,SAAsBU,KAAK,IAAIV,KAAK,CAACU,KAAN,CAAYA,KAAZ,CAA/B;AAAA,CALI,EAME;AAAA,MAAGC,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,IAAIA,KAAK,CAACZ,MAAf,kBAAgCY,KAAK,CAACZ,MAAtC,OAAf;AAAA,CANF,EAQhB;AAAA,MAAGL,UAAH,SAAGA,UAAH;AAAA,MAAeC,aAAf,SAAeA,aAAf;AAAA,SAAmCF,aAAa,CAACC,UAAD,EAAaC,aAAb,CAAhD;AAAA,CARgB,EAShB;AAAA,MAAGiB,MAAH,SAAGA,MAAH;AAAA,SAAgBA,MAAM,yBAAtB;AAAA,CATgB,EAUhB;AAAA,MAAGC,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,IAAIrB,WAAW,CAACqB,KAAD,CAAnC;AAAA,CAVgB,EAYhBxB,KAAK,CAACc,EAZU,EAaI;AAAA,MAAGQ,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,IAAIA,KAAK,CAACP,MAAf,kBAAgCO,KAAK,CAACP,MAAtC,OAAf;AAAA,CAbJ,EAcN;AAAA,MAAGK,MAAH,SAAGA,MAAH;AAAA,SAAgBA,MAAM,IAAIA,MAAM,CAACL,MAAjB,cAA8BK,MAAM,CAACL,MAArC,OAAhB;AAAA,CAdM,EAiBhBf,KAAK,CAACgB,EAjBU,EAkBI;AAAA,MAAGM,KAAH,SAAGA,KAAH;AAAA,SAAeA,KAAK,IAAIA,KAAK,CAACL,OAAf,kBAAiCK,KAAK,CAACL,OAAvC,OAAf;AAAA,CAlBJ,EAmBN;AAAA,MAAGG,MAAH,UAAGA,MAAH;AAAA,SAAgBA,MAAM,IAAIA,MAAM,CAACH,OAAjB,cAA+BG,MAAM,CAACH,OAAtC,OAAhB;AAAA,CAnBM,CAAb","sourcesContent":["import { media } from '@/styles/media'\nimport { css, styled } from '@/styles/styled'\nimport { HeightMap, ImageMap, ThemeColors, VerticalPaddingMap } from '@/types/theme'\nimport { Shape, Corners } from '@/components/Atoms/Card/Card.interface'\nimport { handleShape } from '@/components/Atoms/Card/Card.styled'\n\nexport interface SectionProps {\n  color?: ThemeColors\n  image?: ImageMap | undefined\n  paddingTop?: VerticalPaddingMap\n  paddingBottom?: VerticalPaddingMap\n  height?: HeightMap\n  center?: boolean\n  shape?: Shape | Corners<Shape>\n}\n\nconst handlePadding = (paddingTop, paddingBottom) => {\n  const { ...top } = paddingTop\n  const { ...bottom } = paddingBottom\n\n  return props => css`\n    ${top.mobile && `padding-top: ${props.theme.spacing.base[top.mobile]}`};\n    ${bottom.mobile && `padding-bottom: ${props.theme.spacing.base[bottom.mobile]}`};\n\n    ${media.md} {\n      ${top.tablet && `padding-top: ${props.theme.spacing.base[top.tablet]}`};\n      ${bottom.tablet && `padding-bottom: ${props.theme.spacing.base[bottom.tablet]}`};\n    }\n\n    ${media.lg} {\n      ${top.desktop && `padding-top: ${props.theme.spacing.base[top.desktop]}`};\n      ${bottom.desktop && `padding-bottom: ${props.theme.spacing.base[bottom.desktop]}`};\n    }\n  `\n}\n\nexport const Section = styled.section<SectionProps>`\n  width: 100%;\n  height: ${({ height }) => height && height.mobile && `${height.mobile}px`};\n  left: 0;\n  right: 0;\n  background: ${({ color, theme }) => color && theme.color[color]} no-repeat center center;\n  background-image: ${({ image }) => image && image.mobile && `url(${image.mobile}) `};\n  background-size: cover;\n  ${({ paddingTop, paddingBottom }) => handlePadding(paddingTop, paddingBottom)}\n  ${({ center }) => center && `text-align: center;`}\n  ${({ shape }) => shape && handleShape(shape)}\n\n  ${media.md} {\n    background-image: ${({ image }) => image && image.tablet && `url(${image.tablet}) `};\n    height: ${({ height }) => height && height.tablet && `${height.tablet}px`};\n  }\n\n  ${media.lg} {\n    background-image: ${({ image }) => image && image.desktop && `url(${image.desktop}) `};\n    height: ${({ height }) => height && height.desktop && `${height.desktop}px`};\n  }\n`\n"],"file":"Section.js"}