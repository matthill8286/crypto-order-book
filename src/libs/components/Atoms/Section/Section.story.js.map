{"version":3,"sources":["../../../../src/components/Atoms/Section/Section.story.tsx"],"names":["select","storiesOf","React","saiyanTheme","Section","options","Object","keys","color","knobs","top","tablet","desktop","bottom","mobile","height","bgImage","module","add"],"mappings":";;AAAA,SAASA,MAAT,QAAuB,wBAAvB;AACA,SAASC,SAAT,QAA0B,kBAA1B;AACA,OAAO,KAAKC,KAAZ,MAAuB,OAAvB;AACA,SAASC,WAAT;AAEA,SAASC,OAAT;AAEA,IAAMC,OAAO,GAAGC,MAAM,CAACC,IAAP,CAAYJ,WAAW,CAACK,KAAxB,CAAhB;;AAEA,IAAMC,KAAK,GAAG,SAARA,KAAQ,GAA8B;AAC1C,SAAO;AACLD,IAAAA,KAAK,EAAER,MAAM,CAAc,QAAd,EAAwBK,OAAxB,EAAiC,SAAjC;AADR,GAAP;AAGD,CAJD;;AAMA,IAAMK,GAAuB,GAAG;AAC9BC,EAAAA,MAAM,EAAE,IADsB;AAE9BC,EAAAA,OAAO,EAAE;AAFqB,CAAhC;AAKA,IAAMC,MAA0B,GAAG;AACjCC,EAAAA,MAAM,EAAE,IADyB;AAEjCH,EAAAA,MAAM,EAAE;AAFyB,CAAnC;AAKA,IAAMI,MAAM,GAAG;AACbD,EAAAA,MAAM,EAAE,GADK;AAEbH,EAAAA,MAAM,EAAE,GAFK;AAGbC,EAAAA,OAAO,EAAE;AAHI,CAAf;AAMA,IAAMI,OAAO,GAAG;AACdF,EAAAA,MAAM,EAAE,4CADM;AAEdH,EAAAA,MAAM,EAAE;AAFM,CAAhB;AAKAV,SAAS,CAAC,6BAAD,EAAgCgB,MAAhC,CAAT,CACGC,GADH,CACO,SADP,EACkB,YAAM;AACpB,SACE,oBAAC,OAAD;AAAS,IAAA,UAAU,EAAER,GAArB;AAA0B,IAAA,aAAa,EAAEG;AAAzC,KAAqDJ,KAAK,EAA1D,UADF;AAKD,CAPH,EAQGS,GARH,CAQO,cARP,EAQuB,YAAM;AACzB,SACE,oBAAC,OAAD;AAAS,IAAA,UAAU,EAAER,GAArB;AAA0B,IAAA,aAAa,EAAEG,MAAzC;AAAiD,IAAA,MAAM,EAAEE;AAAzD,KAAqEN,KAAK,EAA1E,UADF;AAKD,CAdH,EAeGS,GAfH,CAeO,YAfP,EAeqB,YAAM;AACvB,SACE,oBAAC,OAAD;AAAS,IAAA,UAAU,EAAER,GAArB;AAA0B,IAAA,KAAK,EAAEM,OAAjC;AAA0C,IAAA,aAAa,EAAEH,MAAzD;AAAiE,IAAA,MAAM,EAAEE;AAAzE,KAAqFN,KAAK,EAA1F,UADF;AAKD,CArBH","sourcesContent":["import { select } from '@storybook/addon-knobs'\nimport { storiesOf } from '@storybook/react'\nimport * as React from 'react'\nimport { saiyanTheme } from '@/styles/sc-vars-saiyan'\nimport { ThemeColors, VerticalPaddingMap } from '@/types/theme'\nimport { Section } from './Section'\n\nconst options = Object.keys(saiyanTheme.color) as ThemeColors[]\n\nconst knobs = (): { color: ThemeColors } => {\n  return {\n    color: select<ThemeColors>('Colors', options, 'primary'),\n  }\n}\n\nconst top: VerticalPaddingMap = {\n  tablet: 'xs',\n  desktop: 'md',\n}\n\nconst bottom: VerticalPaddingMap = {\n  mobile: 'lg',\n  tablet: 'xl',\n}\n\nconst height = {\n  mobile: 350,\n  tablet: 100,\n  desktop: 400,\n}\n\nconst bgImage = {\n  mobile: '/public/images/featured_backgrounds/03.jpg',\n  tablet: '/public/images/featured_backgrounds/01.jpg',\n}\n\nstoriesOf('Design System/Atoms/Section', module)\n  .add('Default', () => {\n    return (\n      <Section paddingTop={top} paddingBottom={bottom} {...knobs()}>\n        Test\n      </Section>\n    )\n  })\n  .add('Fixed Height', () => {\n    return (\n      <Section paddingTop={top} paddingBottom={bottom} height={height} {...knobs()}>\n        Test\n      </Section>\n    )\n  })\n  .add('with Image', () => {\n    return (\n      <Section paddingTop={top} image={bgImage} paddingBottom={bottom} height={height} {...knobs()}>\n        Test\n      </Section>\n    )\n  })\n"],"file":"Section.story.js"}