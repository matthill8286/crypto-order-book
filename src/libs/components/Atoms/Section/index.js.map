{"version":3,"sources":["../../../../src/components/Atoms/Section/index.ts"],"names":[],"mappings":"AAAA","sourcesContent":["export * from './Section'\n"],"file":"index.js"}