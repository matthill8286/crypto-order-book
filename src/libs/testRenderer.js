import { mount } from 'enzyme';
import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { saiyanTheme } from "./styles/sc-vars-saiyan";
export function renderWithTheme(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, tree)).toJSON();
}
export function renderWithThemeAndRouter(tree) {
  var theme = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : saiyanTheme;
  return renderer.create(React.createElement(ThemeProvider, {
    theme: theme
  }, React.createElement(Router, null, tree))).toJSON();
}
export function renderWithRouter(tree) {
  return renderer.create(React.createElement(Router, null, tree)).toJSON();
}
export function mountWithTheme(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, tree));
}
export function mountWithThemeAndRouter(tree) {
  return mount(React.createElement(ThemeProvider, {
    theme: saiyanTheme
  }, React.createElement(Router, null, tree)));
}
//# sourceMappingURL=testRenderer.js.map