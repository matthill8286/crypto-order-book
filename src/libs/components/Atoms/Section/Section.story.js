function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { select } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { saiyanTheme } from "../../../styles/sc-vars-saiyan";
import { Section } from "./Section";
var options = Object.keys(saiyanTheme.color);

var knobs = function knobs() {
  return {
    color: select('Colors', options, 'primary')
  };
};

var top = {
  tablet: 'xs',
  desktop: 'md'
};
var bottom = {
  mobile: 'lg',
  tablet: 'xl'
};
var height = {
  mobile: 350,
  tablet: 100,
  desktop: 400
};
var bgImage = {
  mobile: '/public/images/featured_backgrounds/03.jpg',
  tablet: '/public/images/featured_backgrounds/01.jpg'
};
storiesOf('Design System/Atoms/Section', module).add('Default', function () {
  return React.createElement(Section, _extends({
    paddingTop: top,
    paddingBottom: bottom
  }, knobs()), "Test");
}).add('Fixed Height', function () {
  return React.createElement(Section, _extends({
    paddingTop: top,
    paddingBottom: bottom,
    height: height
  }, knobs()), "Test");
}).add('with Image', function () {
  return React.createElement(Section, _extends({
    paddingTop: top,
    image: bgImage,
    paddingBottom: bottom,
    height: height
  }, knobs()), "Test");
});
//# sourceMappingURL=Section.story.js.map