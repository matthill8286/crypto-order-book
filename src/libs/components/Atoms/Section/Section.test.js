import React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { Section } from "./Section";
describe('Divider', function () {
  it('renders correctly', function () {
    var tree = renderWithTheme(React.createElement(Section, {
      color: "primary"
    }));
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with paddings', function () {
    var tree = renderWithTheme(React.createElement(Section, {
      color: "primary",
      paddingTop: {
        mobile: 'xs',
        tablet: 'lg'
      },
      paddingBottom: {
        mobile: 'lg',
        desktop: 'xs'
      }
    }));
    expect(tree).toMatchSnapshot();
  });
});
//# sourceMappingURL=Section.test.js.map