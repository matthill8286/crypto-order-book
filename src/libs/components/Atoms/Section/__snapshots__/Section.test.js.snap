// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider renders correctly 1`] = `
<section
  className="Section-d5a2e3-0 xlqez"
  color="primary"
/>
`;

exports[`Divider renders correctly with paddings 1`] = `
<section
  className="Section-d5a2e3-0 hKzUdn"
  color="primary"
/>
`;
